src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Services/ReleaseProcessorService.cs

```diff
@@ -75,24 +63,29 @@ private static void CreateReleaseSubjectLink(StatisticsDbContext context, Releas
             }
         }
 
-        private Subject RemoveAndCreateSubject(Guid subjectId, string name, Release release, StatisticsDbContext context)
+        private void RemoveSubject(Guid subjectId, string name, Release release, StatisticsDbContext context)
         {
-            var existingReleaseSubjectLinks = context
-                .ReleaseSubject
-                .Include(r => r.Subject)
-                .Where(r => r.Subject.Name == name && r.ReleaseId == release.Id)
-                .ToList();
-                
-            var existingSubject = existingReleaseSubjectLinks.FirstOrDefault()?.Subject;
-
+            var releaseSubject = context.ReleaseSubject
+                .FirstOrDefault(r => r.Subject.Name == name && r.ReleaseId == release.Id);
```

Comment:
I'm not sure I understand here why we delete the ReleaseSubject identified by Subject Name and later delete the Subject identified by Subject Id.

Reply:
I've changed it to use the name in both cases as I think it's clearer - it did actually work because of the way the restart puts the id into the message but really the subjectId from the message was intended more for seeding.